src/common/interfaces/list.interface.ts

```diff
@@ -5,6 +5,10 @@ export interface List<T> {
     before?: string;
     after?: string;
   };
+  params?: {
```

Comment:
It doesn't seem like this belongs on the `List`, since it's not a value returned from the API.

Reply:
I think this is the crux of auto pagination - this is how Python is currently handling it (adding it to the object). In Node, what's the best way to handle this?

Reply:
We'll need to store the state somewhere else; perhaps on the `AutoPaginatable` class?

We just need to make sure that it's somewhere that can be tracked separately per method call. Since we create a new `AutoPaginatable` instance for each call I think that should be okay.

Reply:
Makes sense - thank you!

Reply:
My tests are definitely going to fail, but maybe somehting like what I have now?